File: squircle.java
import java.util.*;
public class squircle {
public static void main(String[] args)
{
	Scanner input = new Scanner(System.in);
	int T = input.nextInt();
	for(int t = 0; t<T; t++)
	{
		int r = input.nextInt(), s = input.nextInt();
		boolean first = true; // True if we are placing our first (middle) column of square and false otherwise
		double rightX = s/2.0; // The x-coordinate of the rightmost side of the current column of squares
		long res = 0; // The total area we measure
		while(rightX <= r)
		{
			double y = Math.sqrt(r*r - rightX * rightX) * 2; // The height of the circle on the right side of our current column of squares
			int count = (int)(y / s); // The number of squares we can place in the current column
			res += (long)count * s * s * (first ? 1 : 2); // Place count squares, and double it if this is not the middle column due to symmetry.
			rightX += s;
			first = false;
		}
		System.out.println("Measure #" + (t+1)+": "+res);
	}
}
}

File: squircle.py

from math import sqrt, floor

numTests = int(input())

for testCase in range(1, numTests+1):
   radius, sideLength = map(int, input().split())

   # Try placing as many squares as possible at each column.
   multiplier = 1
   numSquares = 0
   position = sideLength * 0.5
   while position < radius:

      # Find the height used using pythagorean theorem.
      # This tells the distance from top to bottom of the circle
      # from the given x location.
      halfHeight = sqrt(radius ** 2 - position ** 2)
      height = 2 * halfHeight

      # Use this height to determine how many squares go here.
      squaresAdded = int(floor(height / sideLength + 1e-9))
      squaresAdded *= multiplier
      numSquares += squaresAdded

      position += sideLength

      # Count both the left and right side after placing the initial column
      multiplier = 2

   # Get the area estimated by the squares
   area = numSquares * sideLength ** 2
   print('Measure #{}: {}'.format(testCase, area))
